app/main.py

```diff
@@ -1,34 +1,212 @@
+                 end: tuple, is_drowned: bool = False) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if start[0] == end[0] and start[1] == end[1]:
+            self.start_column = start[1]
+            self.end_column = self.start_column
+            self.end_row = end[0]
+            self.start_row = self.end_row
+            self.decks.append(Deck(self.start_row, self.end_column))
+            self.position = "single"
+        elif start[0] == end[0]:
+            self.position = "horizontal"
+            row = start[0]
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
+            self.start_column, self.end_column = start[1], end[1]
+        elif start[1] == end[1]:
+            self.position = "vertical"
+            column = start[1]
+            for row in range(start[0], end[0] + 1):
+                self.decks.append(Deck(row, column))
+            self.start_row, self.end_row = start[0], end[0]
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def __str__(self) -> str:
+        ship = ""
+        if self.position == "single":
+            ship += "□"
+        if self.position == "horizontal":
+            for _ in range(len(self.decks)):
+                ship += "□  "
+
+        if self.position == "vertical":
+            for _ in range(len(self.decks)):
+                ship += "□\n"
+
+        return f"\n{ship}\n"
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if (row, column) == (deck.row, deck.column):
+                return deck
+
+    def fire(self, row: int, column: int) -> str:
+        for deck in self.decks:
+            if (row, column) == (deck.row, deck.column):
+                deck.is_alive = False
+        if all(not deck.is_alive for deck in self.decks):
+            self.is_drowned = True
+            return "Sunk!"
+        return "Hit!"
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list) -> None:
+        self.field = {}
+        for ship in ships:
+            new_ship = Ship(ship[0], ship[1])
+            for cell in new_ship.decks:
+                self.field[(cell.row, cell.column)] = new_ship
+        # self._validate_field()
```

Comment:
The `_validate_field` method is commented out. This method is essential for validating the game field and ensuring the correct number of ships. Please uncomment and integrate this method.